src/config/rt/watch.rs

```diff
@@ -80,22 +120,46 @@ impl RtcWatch {
             paths.push(build.target_parent.clone());
         }
 
-        // Take the canonical path of each of the specified ignore targets.
-        let mut ignored_paths = ignore
-            .into_iter()
-            .map(|path| {
-                let path = build.working_directory.join(path);
-                path.canonicalize().map_err(|_| {
-                    anyhow!(
-                        "error taking the canonical path to the watch ignore path: {:?}",
-                        path
-                    )
-                })
-            })
-            .collect::<Result<Vec<_>, _>>()?;
+        let mut ignored_paths = GlobMatcher::new();
 
         // Ensure the final dist dir is always ignored.
-        ignored_paths.push(build.final_dist.clone());
+        let Some(final_dist) = build.final_dist.to_str() else {
+            return Err(anyhow!("could not convert final distribution path to glob"));
+        };
+        let final_dist = globset::Glob::new(final_dist).map_err(|err| anyhow!(err))?;
+        ignored_paths.add(final_dist).map_err(|err| anyhow!(err))?;
+
+        let final_dist_recursive = build.final_dist.join("**");
+        let Some(final_dist_recursive) = final_dist_recursive.to_str() else {
+            return Err(anyhow!("could not convert final distribution path to glob"));
+        };
+        let final_dist_recursive =
+            globset::Glob::new(final_dist_recursive).map_err(|err| anyhow!(err))?;
+        ignored_paths
+            .add(final_dist_recursive)
+            .map_err(|err| anyhow!(err))?;
+
+        let working_dir = build
+            .working_directory
+            .canonicalize()
+            .map_err(|_| anyhow!("error taking the canonical path to the working directory"))?;
+        for path in ignore {
+            let path = working_dir.join(path);
+            let Some(glob) = path.to_str() else {
+                return Err(anyhow!("could not convert {:?} to str", path));
+            };
+            let glob = globset::Glob::new(glob).map_err(|err| anyhow!(err))?;
+            ignored_paths.add(glob).map_err(|err| anyhow!(err))?;
+
+            if !path.is_file() {
```

Comment:
Maybe it makes sense adding some doc here? Or maybe reversing this `if path.is_dir()`?

Reply:
The problem with `path.is_dir` is that if the directory doesn't yet exist then it won't be ignored as expected.

Reply:
Hm, I see. That code is there for the backwards compatibility, right?

If so, why not drop it make a new breaking version?

Reply:
Yeah, it's there for backwards compatibility. 

It seemed that you wanted to [avoid making a breaking change](https://github.com/trunk-rs/trunk/pull/1003#issuecomment-3095793241).

Reply:
Ah, got ya. No worries.